finch.yaml

```diff
@@ -148,27 +148,26 @@ provision:
 - mode: user
   script: |
     #!/bin/bash
+    # We need create /etc/nerdctl directory as root and make it writeable by non root user because the sftp client in https://github.com/runfinch/finch/blob/main/pkg/config/nerdctl_config_applier.go#L140 cannot run as root.
```

Comment:
Adding some additional context, the private key used for authenticating the client to the server(lima instance) is created during `limactl start finch` and the key cannot be used by root user.

Reply:
> Also note thatlimactl [cannot be run as root ](https://github.com/lima-vm/lima/blob/master/cmd/limactl/main.go#L77).

This should refer to `sudo limactl ...` instead of `limactl shell finch sudo ...`, and the latter is what is we want here, so it's a bit unclear why it's relevant here.

> the key cannot be used by root user.

What if we replace these 4 lines with `sudo cp $USER/.ssh/authorized_keys /root/.ssh/authorized_keys` and change the SSH user to `root` [here](https://github.com/runfinch/finch/blob/4a33a2e63c72e687030c70b13dcc5c5748606b86/pkg/config/nerdctl_config_applier.go#L130)?

Reply:
> Also note thatlimactl [cannot be run as root ](https://github.com/lima-vm/lima/blob/master/cmd/limactl/main.go#L77).
What I meant was the `~/.ssh/authorized_keys` is created in the home directory of the non root user as `limactl` can't be run as root. If we could run `sudo limactl` then `authorized_keys` would be placed under `/root/.ssh/`. The suggestion you made below fixes this.
>the key cannot be used by root user.
This is a great idea 👍 , made appropriate changes.

Reply:
> What I meant was the ~/.ssh/authorized_keys is created in the home directory of the non root user as limactl can't be run as root. If we could run sudo limactl then authorized_keys would be placed under /root/.ssh/.

Ah I see! Thank you for the explanation.

> This is a great idea 👍 , made appropriate changes.

Nice! cc @pendo324 as I'm not 100% sure if there's any side effect of doing so.

Reply:
As long as we _copy_ the `authorized_keys` file instead of _moving_ it (which is what your recommendation says), I don't see any issue with this approach. As far as I can tell, the only consequence is that you can SSH into your own VM as your own user, and as root. Since your user can already use `sudo` freely in the VM, I don't see a problem with that.